File: twosumindex.py
def findIndex(nums, target):
    data = {}
    for i in range(len(nums)):
        item1 = nums[i]
        item2 = target - item1
        if data.has_key(item2):
            return [data[item2], i]
        data[item1] = i


print(findIndex([1, 2, 3, 3], 6))
print(findIndex([1, 2, 3, 4], 4))

File: typescript/twosumindex.ts
function findIndex(arr: number[], target: number): number[] {
    let data: {[id: number]: number} = {};
    for(let index=0; index < arr.length; index++) {
        let item1 = arr[index];
        let item2 = target - item1;
        if(typeof data[item2] !== 'undefined') {
            return [data[item2], index];
        }
        data[item1] = index;
    
    }
    return [-1, -1]
}

console.log(findIndex([1, 2, 3, 3,4], 6));
console.log(findIndex([1, 2, 3, 3], 4))